src/mame/atlus/cave.cpp

```diff
@@ -1229,6 +1230,72 @@ void cave_state::paccarn_map(address_map &map)
+      Nissen        |                            |       Nissen
+      Kougeki       |                            |       Kougeki
+ (Heat Ray Attack)  |                            |  (Heat Ray Attack)
+                    |____________________________|
+
+                           ( Blue Speaker )
+
+  To the sides of the screen area there are blue spinning lamps like you'd
+  see atop a police patrol car.
+
+  The buttons are lit when an attack is available. Upon pressing them, the
+  button lamps turn off, while the spinning lamps activate during the attack.
+
+***************************************************************************/
+
+void cave_state::jumbogod_leds_w(u8 data)
+{
+	// Numbering of attack button light outputs is guessed based upon the
+	// ordering of the button inputs in the test menu; in-game they all
+	// appear to be activated at once, so disambiguation is challenging.
+	m_led_outputs[0] = data & 0x0001;  // tai atari / body slam
+	m_led_outputs[1] = data & 0x0002;  // nissen kougeki / atomic breath
+	m_led_outputs[2] = data & 0x0004;  // shippo kougeki / tail attack
+	m_led_outputs[3] = data & 0x0008;  // nissen kougeki / atomic breath
+	m_led_outputs[4] = data & 0x0010;  // motor & spinning lamps
+}
+
+void cave_state::jumbogod_map(address_map &map)
+{
+	map(0x000000, 0x07ffff).rom();                                                                   // ROM
+	map(0x100000, 0x10ffff).ram().share("nvram");                                                    // RAM (battery)
+	map(0x200000, 0x20ffff).ram().share(m_spriteram[0]);                                             // Sprites
+	map(0x300000, 0x307fff).m(m_tilemap[0], FUNC(tilemap038_device::vram_map));                      // Layer 0
+	map(0x400000, 0x40007f).w(FUNC(cave_state::videoregs_w<0>)).share(m_videoregs[0]);               // Video Regs
+	map(0x400000, 0x400007).r(FUNC(cave_state::irq_cause_r));                                        // IRQ Cause
+	map(0x400068, 0x400069).w("watchdog", FUNC(watchdog_timer_device::reset16_w));                   // Watchdog
+	map(0x500000, 0x500005).w(m_tilemap[0], FUNC(tilemap038_device::vregs_w));                       // Layer 0 Control
+	map(0x600000, 0x60ffff).ram().w(m_palette[0], FUNC(palette_device::write16)).share("palette.0"); // Palette
+	map(0x700000, 0x700001).portr("IN0");                                                            // Inputs + EEPROM
+	map(0x700002, 0x700003).portr("IN1");                                                            // Inputs
+	map(0x800001, 0x800001).rw("oki1", FUNC(okim6295_device::read), FUNC(okim6295_device::write));   // M6295
+	map(0x900001, 0x900001).rw("oki2", FUNC(okim6295_device::read), FUNC(okim6295_device::write));   // M6295
+	map(0xc00001, 0xc00001).w(FUNC(cave_state::jumbogod_leds_w));                                    // Leds
+	map(0xe00001, 0xe00001).w(FUNC(cave_state::tjumpman_eeprom_w));                                  // EEPROM
+}   
```

Comment:
This is almost identical to `pacslot_map` – please call `pacslot_map` and then just install the two different handlers you need.

Reply:
Any objection to adding `oki2` to pacslot_map and consolidating the two? I checked and pacslot seems to initialize the second (unused) OKI during startup at the same address. I'd then just have to install the alternate handler for the lamp I/O.

Reply:
Do you know if pacslot has the extra OKI on board?

Reply:
If the second OKI isn’t populated on `pacslot`, it shouldn’t be added.  Many games do run code code to initialise unpopulated peripherals.

Reply:
The peripheral itself is present, it just has no ROM data attached. So, it wouldn't make noise properly if commanded, but it would respond to register reads, etc. It is physically on the PCB on the PCB though.